File: suites/osprey-compute-kernels/reference/cases/pascal/pascal.hs
-- Binomial coefficient via naive (un-memoised) Pascal recursion C(n,k)=C(n-1,k-1)+C(n-1,k).
binom :: Int -> Int -> Int
binom n k
  | k == 0    = 1
  | k == n    = 1
  | otherwise = binom (n - 1) (k - 1) + binom (n - 1) k

main :: IO ()
main = print (binom 27 13)

File: suites/osprey-compute-kernels/reference/cases/pascal/pascal.ml
(* Binomial coefficient via naive (un-memoised) Pascal recursion C(n,k)=C(n-1,k-1)+C(n-1,k). *)
let rec binom n k =
  if k = 0 then 1
  else if k = n then 1
  else binom (n - 1) (k - 1) + binom (n - 1) k

let () = Printf.printf "%d\n" (binom 27 13)
